File: app/src/main/java/com/example/record/model/Category.kt
package com.example.record.model

data class Category (
    val label: String,
    val value: String,
    val children: List<Category>?
)

data class CategoryResponse(
    val default: List<Category>
)

data class CategoryIcon (
    val value: String,
    val icon: String,
    val children: List<CategoryIcon>?
)

data class CategoryIconResponse (
    val default: List<CategoryIcon>
)

File: Record/Record/Model/Category.swift
//
//  Category.swift
//  Record
//
//  Created by xiachuan on 2024/9/3.
//

import Foundation

struct CategoryInfo: Codable {
    var `default`: [Category]
}

struct Category: Codable {
    var label: String
    var value: String
    var children: [Category]?
}

struct CategoryIconInfo: Codable {
    var `default`: [CategoryIcon]
}

struct CategoryIcon: Codable {
    var value: String
    var icon: String
    var children: [CategoryIcon]?
}
